src/libproc_macro/lib.rs

```diff
@@ -499,6 +512,9 @@ impl Literal {
         if !n.is_finite() {
             panic!("Invalid float literal {}", n);
         }
+        if n < 0.0 {
```

Comment:
Looks like negative zeroes still pass the check.

Reply:
Ah oops, just found `is_sign_negative` which is what I was looking for earlier.